Tests/BackendIntegrationTests/OfflineStoreKitIntegrationTests.swift

```diff
@@ -0,0 +1,169 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  OfflineStoreKitIntegrationTests.swift
+//
+//  Created by Nacho Soto on 5/17/23.
+
+import Nimble
+@testable import RevenueCat
+import StoreKit
+import XCTest
+
+class OfflineStoreKit2IntegrationTests: OfflineStoreKit1IntegrationTests {
+
+    override class var storeKit2Setting: StoreKit2Setting { return .enabledForCompatibleDevices }
+
+}
+
+class OfflineStoreKit1IntegrationTests: BaseStoreKitIntegrationTests {
+
+    var serverIsDown: Bool = false
+    override var forceServerErrors: Bool { return self.serverIsDown }
+
+    override func setUp() async throws {
+        try await super.setUp()
+
+        await self.waitForPendingCustomerInfoRequests()
+        if #available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *) {
+            try await self.ensureEntitlementMappingIsAvailable()
+        }
+    }
+
+    func testOfferingsAreCachedInMemory() async throws {
+        let onlineOfferings = try await Purchases.shared.offerings()
+        expect(onlineOfferings.all).toNot(beEmpty())
+
+        self.serverDown()
+
+        let offlineOfferings = try await Purchases.shared.offerings()
+        expect(offlineOfferings) === onlineOfferings
+    }
+
+    func testOfferingsAreCachedOnDisk() async throws {
+        let onlineOfferings = try await Purchases.shared.offerings()
+        expect(onlineOfferings.all).toNot(beEmpty())
+
+        self.serverDown()
+        await self.resetPurchases()
+
+        let offlineOfferings = try await Purchases.shared.offerings()
+        expect(offlineOfferings.response) == onlineOfferings.response
+
+        let offering = try XCTUnwrap(offlineOfferings.current)
+        expect(offering.availablePackages.count) == onlineOfferings.current?.availablePackages.count
+        expect(offering.monthly?.storeProduct.productIdentifier) == "com.revenuecat.monthly_4.99.1_week_intro"
+    }
```

Comment:
I think these offerings tests are not directly related to offline entitlements so not sure if they should be separate... But this is fine for now.

Reply:
It's related to being able to purchase while offline :P